client/src/redux/leagueTeam/reducer.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  setSelectedTeam,
+  setTeamTweets,
+  setTeamFixtures,
+  setFixtureDetails,
+  setSelectedTeamFixtureLastGames,
+} from './thunks';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { controls } from './constants';
+import {
+  Controls,
+  InitialState,
+  Team,
+  TeamTweets,
+  TeamFixtures,
+  TeamFixture,
+  FixtureDetails,
+  DatesFormFields,
+} from './types';
+
+const initialState: InitialState = {
+  selectedTeam: null,
+  selectedControl: controls.players,
+  teamTweets: [],
+  teamFixtures: [],
+  selectedTeamFixtureLastGames: [],
+  selectedTeamFixture: null,
+  fixtureDetails: null,
+  teamFixturesError: '',
+  tweetsQuery: '',
+  datesFormFields: null,
+};
+
+export const leagueTeamSlice = createSlice({
+  name: 'leagueTeam',
+  initialState,
+  reducers: {
+    setSelectedControl: (state, action: PayloadAction<Controls>) => {
+      state.selectedControl = action.payload;
+    },
+    setTweetsQuery: (state, action: PayloadAction<string>) => {
+      state.tweetsQuery = action.payload;
+    },
+    setSelectedTeamFixture: (state, action: PayloadAction<TeamFixture>) => {
+      state.selectedTeamFixture = action.payload;
+    },
+    setDatesFormFields: (state, action: PayloadAction<DatesFormFields>) => {
+      state.datesFormFields = action.payload;
+    },
+  },
+  extraReducers: {
+    [setSelectedTeam.fulfilled.type]: (state, action: PayloadAction<Team>) => {
```

Comment:
Does these lines also breaks everything because of cycle dependency issue or similar?

Reply:
Nope.